library/alloc/src/collections/binary_heap.rs

```diff
@@ -519,78 +526,135 @@ impl<T: Ord> BinaryHeap<T> {
     // the hole is filled back at the end of its scope, even on panic.
     // Using a hole reduces the constant factor compared to using swaps,
     // which involves twice as many moves.
-    fn sift_up(&mut self, start: usize, pos: usize) -> usize {
-        unsafe {
-            // Take out the value at `pos` and create a hole.
-            let mut hole = Hole::new(&mut self.data, pos);
-
-            while hole.pos() > start {
-                let parent = (hole.pos() - 1) / 2;
-                if hole.element() <= hole.get(parent) {
-                    break;
-                }
-                hole.move_to(parent);
+
+    /// # Safety
+    ///
+    /// The caller must guarantee that `pos < self.len()`.
```

Comment:
Why do we need an extra space and make this a doc block?

Reply:
The extra space is there because the upper comment is not exclusive to `sift_up` but it's shared with `sift_down_range` and `sift_down_to_botton` too. 

The lower comment is a doc comment to be consistent with the other private methods since they're also have doc comments.